pkg/metrics/metrics.go

```diff
@@ -212,3 +208,16 @@ func EmitCSVMetric(oldCSV *olmv1alpha1.ClusterServiceVersion, newCSV *olmv1alpha
 		csvAbnormal.WithLabelValues(newCSV.Namespace, newCSV.Name, newCSV.Spec.Version.String(), string(newCSV.Status.Phase), string(newCSV.Status.Reason)).Set(1)
 	}
 }
+
+func EmitSubsMetrics(sub *olmv1alpha1.Subscription) {
+	// sub.Spec is not a required field.
+	if sub.Spec == nil {
+		return
+	}
+
+	SubscriptionSyncCount.WithLabelValues(sub.GetName(), sub.Status.InstalledCSV, sub.Spec.Channel, sub.Spec.Package).Inc()
```

Comment:
Note: sub.Spec.Channel and sub.Spec.Package are not required fields - these could be empty strings. Should we emit these metrics?

Reply:
@anik120 this is the only non-nit that I have on the PR, great work.

Reply:
@awgreene what is the issue if we emit empty string?

Reply:
Should we waste telemeter bandwith reporting a sub that doesn't have:

- A name
- A CSV that it installs
- Define a package or channel to pull from?

Reply:
@awgreene  isn't name, installedCSV required fields?

Reply:
>If the Subscription is being created on cluster successfully, I'm not convinced we should choose tweak the metrics to not emit data point based on field values.
That would essentially mean we have subscriptions on cluster, but telemeter is unaware of them.

If we can use those incomplete metrics we omit to find identify something useful, then yes, we should emit them. Otherwise, we're just consuming our telemeter quota for no reason.

Reply:
@awgreene so in summary: 

a) We could either not emit those metrics now, and if we run into a use case where we need those data points we can start emitting them too. But then they'd have to be backported into previous z stream releases (which we already know is an EXTREMELY delayed process, compared to how rapidly telemeter dashboards can be created)

b) We could just emit the metric since a subscription is being created (and note that it'll be deleted now once the subscription is deleted) which will use up some of the telemeter quota. Question here is: how often are we expecting these kind of subscriptions to be created (my guess is they won't be created in any prod clusters)? And as a result, how much quota are we expecting to use up for this edge case? 

I just want to make sure we don't put our cart before our horses and potentially expose ourselves to a bug back porting process.

Reply:
I am unconvinced that there will ever be a point that we will want to consume the metric when the installedCSV is blank.

However, given that the metrics are deleted when the subscription is deleted, I see no way that the metric could blow up. I also agree with your point about the ease of updating telemeter vs backporting changes. Given those points, it probably isn't worth delaying this bug fix.